dbms/src/AggregateFunctions/AggregateFunctionGroupUniqArray.h

```diff
@@ -154,9 +150,21 @@ class AggregateFunctionGroupUniqArrayGeneric
 
     using State = AggregateFunctionGroupUniqArrayGenericData;
 
-    static StringRef getSerialization(const IColumn & column, size_t row_num, Arena & arena)
+    using KeyHolder = typename std::conditional<is_plain_column,
+        ArenaKeyHolder, SerializedKeyHolder>::type;
+
+    static KeyHolder getKeyHolder(const IColumn & column, size_t row_num, Arena & arena)
     {
-        return getSerializationImpl<is_plain_column>(column, row_num, arena);
+        if constexpr (is_plain_column)
+        {
+            return ArenaKeyHolder{column.getDataAt(row_num), arena};
+        }
+        else
+        {
+            const char * begin = nullptr;
```

Comment:
we can leave out this var

Reply:
It's an out-parameter, so we have to keep it.